src/diffusers/models/attention_processor.py

```diff
@@ -303,6 +304,22 @@ def set_attention_slice(self, slice_size):
         self.set_processor(processor)
 
     def set_processor(self, processor: "AttnProcessor"):
+        if (
+            hasattr(self, "processor")
+            and not isinstance(processor, LORA_ATTENTION_PROCESSORS)
+            and self.to_q.lora_layer is not None
+        ):
```

Comment:
Neat. 

I am a little concerned about:

```
and self.to_q.lora_layer is not None
```

i.e., only check if `to_q.lora_layer` is not None. Is there a better alternative?